src/core/pointcloud/qgscopcpointcloudindex.cpp

```diff
@@ -43,6 +43,35 @@ QgsCopcPointCloudIndex::QgsCopcPointCloudIndex() = default;
 
 QgsCopcPointCloudIndex::~QgsCopcPointCloudIndex() = default;
 
+QgsPointCloudIndex *QgsCopcPointCloudIndex::clone() const
+{
+  QgsCopcPointCloudIndex *clone = new QgsCopcPointCloudIndex;
+  QMutexLocker locker( &mHierarchyMutex );
+
+  // Base QgsPointCloudIndex fields
+  clone->mExtent = mExtent;
+  clone->mZMin = mZMin;
+  clone->mZMax = mZMax;
+  clone->mHierarchy = mHierarchy;
+  clone->mScale = mScale;
+  clone->mOffset = mOffset;
+  clone->mRootBounds = mRootBounds;
+  clone->mAttributes = mAttributes;
+  clone->mSpan = mSpan;
+  clone->mFilterExpression = mFilterExpression;
+
+  // QgsCopcPointCloudIndex specific fields
+  clone->mIsValid = mIsValid;
+  clone->mFileName = mFileName;
+  clone->mCopcFile.open( mFileName.toStdString(), std::ios::binary );
+  clone->mCopcInfoVlr = mCopcInfoVlr;
+  clone->mHierarchyNodePos = mHierarchyNodePos;
+  clone->mOriginalMetadata = mOriginalMetadata;
```

Comment:
Can this itself be cloned?

Reply:
Sorry I didn't get what you mean, can you explain more?

Reply:
I mean that QgsLazInfo does a lot of work to initialise, including remote requests in some cases. Could we avoid that by adding a clone method to QgsLazInfo itself too and copying what we already know directly (instead of regarding it all).

Reply:
I addressed that in the latest commit, thanks for the clarification!